receiver/k8seventsreceiver/k8s_event_to_logdata.go

```diff
@@ -0,0 +1,97 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8seventsreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/k8seventsreceiver"
+
+import (
+	"fmt"
+	"strings"
+
+	"go.opentelemetry.io/collector/model/pdata"
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	// Number of log attributes to add to the pdata.LogSlice.
+	totalLogAttributes = 9
+
+	// Number of resource attributes to add to the pdata.ResourceLogs.
+	totalResourceAttributes = 3
+)
+
+// Only two types of events are created as of now.
+// For more info: https://docs.openshift.com/container-platform/4.9/rest_api/metadata_apis/event-core-v1.html
+var severityMap = map[string]pdata.SeverityNumber{
+	"normal":  pdata.SeverityNumberINFO,
+	"warning": pdata.SeverityNumberWARN,
+}
+
+// k8sEventToLogRecord converts Kubernetes event to pdata.LogSlice and adds the resource attributes.
+func k8sEventToLogData(logger *zap.Logger, ev *corev1.Event) pdata.Logs {
+	ld := pdata.NewLogs()
+	rl := ld.ResourceLogs().AppendEmpty()
+	ill := rl.InstrumentationLibraryLogs().AppendEmpty()
+	lr := ill.Logs().AppendEmpty()
+
+	resourceAttrs := rl.Resource().Attributes()
+	resourceAttrs.Clear()
+	resourceAttrs.EnsureCapacity(totalResourceAttributes)
+
+	resourceAttrs.InsertString("k8s.cluster.name", ev.ObjectMeta.ClusterName)
+	resourceAttrs.InsertString("k8s.event.source", ev.Source.Component)
+	resourceAttrs.InsertString("k8s.node.name", ev.Source.Host)
+
+	// The Reason field is the most logical "name" of the event.
+	lr.SetName(ev.Reason)
+	lr.SetTimestamp(pdata.Timestamp(getEventTimestamp(ev).UnixNano()))
+
+	// The Message field contains description about the event,
+	// which is best suited for the "Body" of the LogSlice.
+	lr.Body().SetStringVal(ev.Message)
+
+	// Set the "SeverityNumber" and "SeverityText" if a known type of
+	// severity is found.
+	if severityNumber, ok := severityMap[strings.ToLower(ev.Type)]; ok {
+		lr.SetSeverityNumber(severityNumber)
+		lr.SetSeverityText(ev.Type)
+	} else {
+		logger.Debug("unknown severity type", zap.String("type", ev.Type))
+	}
+
+	attrs := lr.Attributes()
+	attrs.Clear()
+	attrs.EnsureCapacity(totalLogAttributes)
+
+	attrs.InsertString("k8s.event.action", ev.Action)
+	attrs.InsertString("k8s.event.start_time", ev.ObjectMeta.CreationTimestamp.String())
+	attrs.InsertString("k8s.event.name", ev.ObjectMeta.Name)
+	attrs.InsertString("k8s.event.uid", string(ev.ObjectMeta.UID))
+	attrs.InsertString("k8s.container.name", ev.InvolvedObject.Name)
```

Comment:
Does `ev.InvolvedObject.Name` always represent a container name? I though it can be `k8s.pod.name` as well, no?

Reply:
It is `ev.InvolvedObject.FieldPath` which represents container name and I have updated it.

Reply:
Is `ev.InvolvedObject.FieldPath` always has a container name? May it's true only when ev.InvolvedObject is a pod?

Reply:
Yes I think  `k8s.object.fieldpath` can be used for now.

For the next version of the receiver we should provide an option to extend all the raw attributes, e.g. `k8s.object.kind+k8s.object.name` -> `k8s.<pod|...>.name`, and extraction of the `k8s.container.name` value.

Reply:
I have added the convention to the same PR and also used the same in this PR.